src/main/java/seedu/hirehub/model/person/ContainsKeywordsPredicate.java

```diff
@@ -56,7 +74,22 @@ public boolean test(Person person) {
             }
             return isPersonTagsContainsTag(person, tags);
         }
-        throw new IllegalStateException("Unexpected value: " + prefix);
+        throw new IllegalStateException("Unexpected prefix: " + prefix);
+    }
+
+    private boolean applicationTest(Application application) {
+        if (prefix.equals(PREFIX_EMAIL)) {
+            Optional<Email> email = (Optional<Email>) keywords;
+            return application.getPerson().getEmail().value
+                    .equals(email.orElse(application.getPerson().getEmail()).value);
```

Comment:
also isn't this always true if user does not provide any keyword? So that means if user provides empty input (e.g. e/ ), then the search output is true is it?

Reply:
The SearchApplicationCommandParser would parse the email and finds that the empty input is an invalid email. It would throw a ParseException, so it would never reach here. But you do raise a good point, I should probably add assertions to prevent misuse of this code.

Reply:
Okie noted